web-frontend/src/pages/Home/Home.tsx

```diff
@@ -1,40 +1,63 @@
-import { useParams } from 'react-router-dom';
+import { useNavigate, useParams } from 'react-router-dom';
 import { AccountObject } from '../../models/AccountModel';
-import { formatStandard, Unit } from '../../utils/MassaFormating';
+import { formatStandard, Unit, maskAddress } from '../../utils/MassaFormating';
 import { useResource } from '../../custom/api';
+import { routeFor } from '../../utils';
 import Intl from '../../i18n/i18n';
 
 import WalletLayout, {
   MenuItem,
 } from '../../layouts/WalletLayout/WalletLayout';
-import { Button, Balance } from '@massalabs/react-ui-kit';
-import { FiArrowDownLeft, FiArrowUpRight } from 'react-icons/fi';
+import { Button, Balance, Input } from '@massalabs/react-ui-kit';
+import { FiArrowDownLeft, FiArrowUpRight, FiCopy } from 'react-icons/fi';
 
 export default function Home() {
+  const navigate = useNavigate();
   const { nickname } = useParams();
   const { data: account } = useResource<AccountObject>(`accounts/${nickname}`);
 
-  const unformattedBalance = account?.candidateBalance ?? '0';
+  if (account === undefined) {
+    navigate(routeFor(`index`));
+    return null;
+  }
```

Comment:
We are relying in the wrong hook. This is not the way. I'll show you just after the daily.

Reply:
Yes! Let we sync to solve this problem.